share/spack/setup-env.sh

```diff
@@ -189,19 +186,65 @@ if [ -z "$_sp_source_file" ]; then
 fi
 
 #
-# Set up modules and dotkit search paths in the user environment
+# Find root directory and add bin to path.
 #
 _sp_share_dir=$(cd "$(dirname $_sp_source_file)" && pwd)
 _sp_prefix=$(cd "$(dirname $(dirname $_sp_share_dir))" && pwd)
 _spack_pathadd PATH       "${_sp_prefix%/}/bin"
+export SPACK_ROOT=${_sp_prefix}
+
+#
+#
+# Determine which shell is being used
+#
+function _spack_determine_shell() {
+	ps -p $$ | tail -n 1 | awk '{print $4}' | xargs basename
+}
+export SPACK_SHELL=$(_spack_determine_shell)
+
+#
+# Check whether at least one of 'use' and 'module' exists
+#
+function _spack_fn_exists() {
+	type $1 2>&1 | grep -q 'shell function'
+}
+
+need_module="no"
+if [ ! $(_spack_fn_exists use) ]; then
+	if [ ! $(_spack_fn_exists module) ]; then
+		need_module="yes"
+	fi;
+fi;
 
+#
+# build and make available environment-modules
+#
+if [ "${need_module}" = "yes" ]; then
+    #check if environment-modules~X is installed
+    module_prefix="$(spack location -i "environment-modules~X" 2>&1 || echo "not_installed")"
```

Comment:
If a user takes the manual step of calling ```spack install environment-modules``` then this will not detect it (since that installs w/ ```+X```). I think you can just search for ```environment-modules```

Reply:
Yep, you're right changed it.